src/platform/Tizen/ConnectivityUtils.cpp

```diff
@@ -72,14 +98,602 @@ app::Clusters::GeneralDiagnostics::InterfaceTypeEnum ConnectivityUtils::GetInter
         Platform::CopyString(ifr.ifr_name, ifname);
 
         if (ioctl(sock, SIOCETHTOOL, &ifr) != -1)
-            ret = app::Clusters::GeneralDiagnostics::InterfaceTypeEnum::EMBER_ZCL_INTERFACE_TYPE_ENUM_ETHERNET;
+            ret = InterfaceTypeEnum::EMBER_ZCL_INTERFACE_TYPE_ENUM_ETHERNET;
     }
 
     close(sock);
 
     return ret;
 }
 
+CHIP_ERROR ConnectivityUtils::GetInterfaceHardwareAddrs(const char * ifname, uint8_t * buf, size_t bufSize)
+{
+    CHIP_ERROR err = CHIP_ERROR_READ_FAILED;
+    int skfd;
+
+    if ((skfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0)
+    {
+        ChipLogError(DeviceLayer, "Failed to create a channel to the NET kernel.");
+        return CHIP_ERROR_OPEN_FAILED;
+    }
+
+    if (ifname[0] != '\0')
```

Comment:
Now, if `ifname == ""` this function will return CHIP_ERROR_READ_FAILED, maybe better it would be to return `CHIP_ERROR_READ_FAILED` in that case?

Reply:
Invalid argument error seems to be more appropriate. i'll update it.